cloud/aws/cloudwatchlogs/getLogGroup.go

```diff
@@ -0,0 +1,63 @@
+package cloudwatchlogs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
+)
+
+func GetLogGroup(ctx context.Context, client *cloudwatchlogs.Client, req GetLogGroupRequest) GetLogGroupResponse {
+	request := &cloudwatchlogs.DescribeLogGroupsInput{
+		LogGroupNamePrefix: aws.String(req.Name),
+	}
+	response, err := client.DescribeLogGroups(ctx, request)
+	if err != nil {
+		return GetLogGroupResponse{
+			error: err,
+		}
+	}
+
+	logGroup := getLogGroupByName(response, req.Name)
+	if logGroup == nil {
+		return GetLogGroupResponse{
+			response: response,
+			error:    fmt.Errorf("LogGroup not found with name %v", req.Name),
+		}
+	}
+
+	return GetLogGroupResponse{
+		logGroup: logGroup,
+		response: response,
```

Comment:
I understand that the AWS  client doesn't provide us with a way to obtain a single log group (filtering by name), right?   I see in the code that we can request the log groups that have the same prefix and that's the reason why we need to iterate later over the log groups and take the one whose name match the name in the request.  If I am right I think that we can omit the attribute response from the the `GetLogGroupResponse` struct since we are not interested at all in the log groups whose name starts by a given prefix.  Actually the signature of the method is `GetLogGroup`